src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```diff
@@ -0,0 +1,522 @@
+        final String[] recordForOneBreakPoint = recordLine.split("\t");
+        final Long breakpointID = Long.valueOf(recordForOneBreakPoint[0]);
+        try{
+            final URI fastqURI = new URI(recordForOneBreakPoint[1]);
+
+            return new Tuple2<>(breakpointID, fastqURI);
+        }catch (final Exception e){
+            return new Tuple2<>(breakpointID, null);
+        }
+    }
+
+    /**
+     * Validates the returned result from running the local assembly pipeline:
+     *   if all steps executed successfully, the contig file is nonnull so we save the contig file and discard the runtime information
+     *   if any step returns non-zero code, the contig file is null so we save the runtime information for that break point
+     * @param results       the local assembly result and its associated breakpoint ID
+     * @param outputDir     output directory to save the contigs (if assembly succeeded) or runtime info (if erred)
+     */
+    private static void validateAndSaveResults(final JavaPairRDD<Long, SGAAssemblyResult> results, final String outputDir){
+        results.cache(); // cache because Spark doesn't have an efficient RDD.split(predicate) yet
+        final JavaPairRDD<Long, SGAAssemblyResult> success = results.filter(entry -> entry._2().assembledContigs!=null);
+        success.mapToPair(entry -> new Tuple2<>(entry._1(), entry._2().assembledContigs))
+                .saveAsObjectFile(outputDir);
+        final JavaPairRDD<Long, SGAAssemblyResult> failure = results.filter(entry -> entry._2().assembledContigs==null);
+        failure.mapToPair(entry -> new Tuple2<>(entry._1(), entry._2().runtimeInformation))
+                .saveAsObjectFile(outputDir);
+    }
+
+    /**
+     * Final return type of the whole process of SGA local assembly.
+     * assembledContigFile is the file containing the assembled contigs, if the process executed successfully, or null if not.
+     * runtimeInformation contains the runtime information logged along the process up until the process erred, if errors happen,
+     *  or until the last step if no errors occur along the line.
+     *  The list is organized along the process of executing the assembly pipeline, inside each entry of the list is
+     *      the name of the module,
+     *      return status of the module,
+     *      stdout message of the module, if any,
+     *      stderr message of the module, if any
+     */
+    @VisibleForTesting
+    static final class SGAAssemblyResult implements Serializable{
+        private static final long serialVersionUID = 1L;
+
+        public final ContigsCollection assembledContigs;
+        public final List<Tuple2<String, Tuple3<ExternalCommandlineProgramModule.ReturnStatus, String, String>>> runtimeInformation;
+
+        public SGAAssemblyResult(final File contigFile, final List<Tuple2<String, Tuple3<ExternalCommandlineProgramModule.ReturnStatus, String, String>>> collectedRuntimeInfo){
```

Comment:
To me it would be nicer if this method took as parameters the return status, stdin, and stdout, rather than a Tuple3. Within this constructor you can pack them into a Tuple3 but otherwise it'll be hard for someone reading the signature of the method to know what the entries of the Tuple3 should be.